includes/utilities.inc

```diff
@@ -356,7 +357,32 @@ function islandora_solr_prepare_solr_results($solr_results) {
       elseif (isset($highlighting[$pid][$field])) {
         $value = $highlighting[$pid][$field];
       }
-
+      // Replace PID with Object Label.
+      if (in_array($field, $replace_pid_with_label)) {
+        $query_processor = new IslandoraSolrQueryProcessor();
+        $label = NULL;
+        $pid_to_label_mapper = function ($uri) use ($query_processor) {
+          $pid = str_replace('info:fedora/', '', $uri);
+          if (islandora_is_valid_pid($pid)) {
+            $query_processor->buildQuery("PID:\"$pid\"");
+            $query_processor->solrParams['fl'] = 'PID, ' . variable_get('islandora_solr_object_label_field', 'fgs_label_s');
+            $query_processor->executeQuery();
+            if (!empty($query_processor->islandoraSolrResult) && !empty($query_processor->islandoraSolrResult['response']['objects'])) {
```

Comment:
Just a comment, but this posse (maybe, correct me if wrong) the following issue (and the object load part also). What happens if the current user has no access/is not allowed to see the object that is referred in the PID/URI?  
What happens if the object is actually not there anymore? (orphan who lost parent)

Reply:
Haha want to find out? I am guessing it's not good things. Nice catch, Diego. Thanks!

Reply:
With commit 0790823:  say you have an object appearing in search results (islandora:123) and it contains, in one of its solr fields, a reference to an inaccessible object (secret:999, with label "My secret object"). 

If secret:999 is not in an accessible namespace, and solr is configured to block that namespace, then you should now see "secret:999". If solr allows that namespace because you configured solr inconsistently from islandora, then you see "My secret object".

If secret:999 is blocked by XACML then it should not appear in an islandoraSolrResult (because the filter of 

```RELS_EXT_isViewableByUser_literal_ms:"admin" OR RELS_EXT_isViewableByRole_literal_ms:"authenticated user" OR RELS_EXT_isViewableByRole_literal_ms:"administrator" OR ((*:* -RELS_EXT_isViewableByUser_literal_ms:[* TO *]) AND (*:* -RELS_EXT_isViewableByRole_literal_ms:[* TO *]))``` 

gets added to the islandoraSolrQueryProcessor), so you see "secret:999". 

However, this brings to light another issue: in the Solr Settings > Solr Query Defaults screen, there's a Solr base filter. You might use this to prevent collections or other types of objects from appearing in solr results. The description is "Lists base filters that are appended to all *user* queries" (emphasis mine). That base filter is also applied to these queries, so if you're blocking, say, collections, from appearing in the user's search results, then you're also blocking them from appearing in internal calls such as this. I think this is an issue, so I'll add it to Jira if it's not there already.

Reply:
@rosiel so the issue is that you have (for example)
`resourceA  example:hasRelated resourceB`

I can access `resourceA` but not `resourceB` so by viewing `resourceA` I can see that `resourceB` exists, but you don't want the link to appear at all.

For Solr is sounds like we'd need to know which fields contain a reference to another resource and to loop through those fields after resolving the query to try to filter out the "hidden" links.

Not sure if there is an easy way to do that.

Reply:
@whikloj I was taking a simpler approach: that "example:hasRelated info:fedora/resourceB" is part of resourceA's metadata, and therefore if you can see resourceA you can see that relationship, and resourceB's URI and/or PID. But if you're not allowed to see resourceB then you cannot resolve its URI to its label. 

Does that sound reasonable?

Reply:
Sorry I must have read your comment wrong, I thought you did not like that as the solution. If you are happy, then I'm happy. 😄